### ISSUE TITLE
Build: Eco Staging build failing to create new default world

### ISSUE THREAD

**Issue by redwyre** [StrangeLoopGames/EcoIssues#17465]
It seems like sometimes the Eco Staging build fails to save the world when it's generating a new one to replace DefaultWorld

**Comment by redwyre** [comment]
Hasn't failed for a while so I'll close this

**Comment by redwyre** [comment]
Ok this started happening again

I'm not sure why it is happening, it just looks like the server isn't saving the game.eco file

**Comment by mirasrael** [comment]
It seems it causes by problems with the building node I/O:
```
DEBUG:   16+  >>>> Push-Location "$ProjectDir\NewDefault"
DEBUG:   18+  >>>> Remove-Item DefaultWorld -ErrorAction SilentlyContinue
DEBUG:   19+  >>>> Start-Process -FilePath "EcoServer.exe" -Args "-generateDefault" -Wait
Cannot contact friend rice: java.lang.InterruptedException
DEBUG:   22+  >>>> Start-Sleep -Seconds 5
DEBUG:   24+  >>>> Copy-Item "Storage\Game.eco" -Destination "$ProjectDir\Build\Server\DefaultWorld" -Force
```

this line: `Cannot contact friend rice: java.lang.InterruptedException`

https://stackoverflow.com/questions/46200130/jenkins-build-fails-on-slaves-with-java-lang-interruptedexception

**Comment by redwyre** [comment]
This is the log I normally get, I think that exception was unrelated

```
Generating default world
[Pipeline] powershell
DEBUG:    5+  >>>> $ScriptDir = Split-Path -Parent -Path $MyInvocation.MyCommand.Definition
DEBUG:    6+  >>>> $ProjectDir = Resolve-Path "${ScriptDir}\..\"
DEBUG:    9+ if ( >>>> Test-Path "$ProjectDir\NewDefault") {
DEBUG:    9+ if ( >>>> Test-Path "$ProjectDir\NewDefault") {
DEBUG:   13+  >>>> Copy-Item -Path "$ProjectDir\Build\Server\win-x64\" -Destination "$ProjectDir\NewDefault\" -Recurse
DEBUG:   16+  >>>> Push-Location "$ProjectDir\NewDefault"
DEBUG:   18+  >>>> Remove-Item DefaultWorld -ErrorAction SilentlyContinue
DEBUG:   19+  >>>> Start-Process -FilePath "EcoServer.exe" -Args "-generateDefault" -Wait
DEBUG:   21+  >>>> Copy-Item "Storage\Game.eco" -Destination "$ProjectDir\Server\DefaultWorld"
powershell.exe : Copy-Item : Cannot find path 'C:\Jenkins\workspace\Eco Release\NewDefault\Storage\Game.eco' because it does not exist.
At C:\Jenkins\workspace\Eco Release@tmp\durable-bbeb1f38\powershellWrapper.ps1:3 char:1
+ & powershell -NoProfile -NonInteractive -ExecutionPolicy Bypass -Comm ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (Copy-Item : Can...does not exist.:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
At C:\Jenkins\workspace\Eco Release\Tools\GenerateDefaultWorld.ps1:21 char:1

+ Copy-Item "Storage\Game.eco" -Destination "$ProjectDir\Server\Default ...

+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

    + CategoryInfo          : ObjectNotFound: (C:\Jenkins\work...torage\Game.eco:String) [Copy-Item], ItemNotFoundExce 

   ption

    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.CopyItemCommand
```

**Comment by mirasrael** [comment]
I reproduced this locally when had local Server already started. It failed to start because of port conflicts. I guess this issue happens when another build in progress which runs integration tests.
I refactored the script to output the server log to console for troubleshooting and also supported net-less mode for generate default world and migration tests, so they won't overlap with other servers running on build agent.
